src/go/cmd/http-relay-client/main.go

```diff
@@ -563,7 +645,7 @@ func main() {
 			}
 		}
 
-		transport = h2transport
+		transport = &ochttp.Transport{Base: h2transport}
```

Comment:
move `&ochttp.Transport{Base: ` to where we set the transport: `Transport: transport,`